File: algorithm/stringproblem/threeSum_test.go
package stringproblem

import (
	"reflect"
	"testing"
)

func Test_threeSum(t *testing.T) {
	type args struct {
		nums []int
	}
	tests := []struct {
		name string
		args args
		want [][]int
	}{
		{name: "simple",
			args: args{nums: []int{-1, 0, 1, 2, -1, -4}},
			want: [][]int{{-1, -1, 2}, {-1, 0, 1}},
		},
		{name: "duplication",
			args: args{nums: []int{-1, 1, -1, 1, -1, 0}},
			want: [][]int{{-1, 0, 1}},
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := threeSum2(tt.args.nums); !reflect.DeepEqual(got, tt.want) {
				t.Errorf("threeSum() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: algorithm/array-sort/P15ThreeSum/threeSum.test.js
const threeSum = require('./threeSum');

test('simple', () => {
  expect(threeSum([-1, 0, 1, 2, -1, -4])).toEqual([[-1, -1, 2], [-1, 0, 1]]);
});

test('duplication', () => {
  expect(threeSum([1, 1, -1, 1, -1, 0])).toEqual([[-1, 0, 1]]);
});
